versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTableBackend.java

```diff
@@ -0,0 +1,212 @@
+        ex = new RuntimeException(e);
+      }
+      try {
+        if (tableAdminClient != null) {
+          tableAdminClient.close();
+        }
+      } catch (Exception e) {
+        if (ex == null) {
+          ex = new RuntimeException(e);
+        } else {
+          ex.addSuppressed(e);
+        }
+      }
+      if (ex != null) {
+        throw ex;
+      }
+    }
+  }
+
+  @Override
+  public void setupSchema() {
+    if (tableAdminClient == null) {
+      // If BigTable admin client is not available, check at least that the required tables exist.
+      boolean refs = checkTableNoAdmin(TABLE_REFS);
+      boolean objs = checkTableNoAdmin(TABLE_OBJS);
+      checkState(
+          refs && objs,
+          "Not all required tables (%s and %s) are available in BigTable, cannot start.",
+          TABLE_REFS,
+          TABLE_OBJS);
+      LOGGER.info("No Bigtable admin client available, skipping schema setup");
+      return;
+    }
+
+    checkTable(TABLE_REFS, FAMILY_REFS);
+    checkTable(TABLE_OBJS, FAMILY_OBJS);
+  }
+
+  private boolean checkTableNoAdmin(String table) {
+    try {
+      dataClient.readRow(table, "dummy");
+      return true;
+    } catch (NotFoundException nf) {
+      LOGGER.error("Nessie table '{}' does not exist in Google Bigtable", table);
+    }
+    return false;
+  }
+
+  private void checkTable(String table, String family) {
+    BigtableTableAdminClient client = requireNonNull(tableAdminClient);
+    try {
+      client.getTable(table);
+    } catch (NotFoundException nf) {
+      LOGGER.info("Creating Nessie table '{}' in Google Bigtable...", table);
+      client.createTable(CreateTableRequest.of(table).addFamily(family));
+    }
+  }
+
+  @Override
+  public void eraseRepositories(Set<String> repositoryIds) {
+    if (!eraseRepositoriesAdminClient(repositoryIds)) {
+      eraseRepositoriesNoAdminClient(repositoryIds);
+    }
+  }
+
+  private boolean eraseRepositoriesAdminClient(Set<String> repositoryIds) {
+    if (tableAdminClient == null) {
+      return false;
+    }
+
+    for (String repoId : repositoryIds) {
+      ByteString prefix = copyFromUtf8(repoId + ':');
+      tableAdminClient.dropRowRange(TABLE_REFS, prefix);
+      tableAdminClient.dropRowRange(TABLE_OBJS, prefix);
+    }
+
+    return true;
+  }
+
+  private void eraseRepositoriesNoAdminClient(Set<String> repositoryIds) {
+    List<ByteString> prefixes =
+        repositoryIds.stream()
+            .map(repoId -> copyFromUtf8(repoId + ':'))
+            .collect(Collectors.toList());
+    eraseRepositoriesTable(TABLE_REFS, prefixes);
+    eraseRepositoriesTable(TABLE_OBJS, prefixes);
+  }
+
+  private void eraseRepositoriesTable(String tableId, List<ByteString> prefixes) {
+    BulkMutation bulkDelete = BulkMutation.create(tableId);
+
+    Query.QueryPaginator paginator = Query.create(tableId).createPaginator(100);
+    ByteString lastKey;
+    Iterator<Row> rows = dataClient.readRows(paginator.getNextQuery()).iterator();
```

Comment:
Since we only care about the row key, you can add a filter:

```java
    Iterator<Row> rows =
        dataClient
            .readRows(paginator.getNextQuery().filter(Filters.FILTERS.limit().cellsPerRow(1)))
            .iterator();
```